### PR TITLE
add tests for syntax

### PR DESCRIPTION
**Pull request by phase06**
## Type of changes

- [ ] Bug fix
- [ ] New feature
- [ ] Documentation / docstrings
- [x] Tests
- [ ] Other

## Checklist

- [x] I've run the latest [black](https://github.com/psf/black) with default args on new code.
- [ ] I've updated CHANGELOG.md and CONTRIBUTORS.md where appropriate.
- [x] I've added tests for new code.
- [x] I accept that @willmcgugan may be pedantic in the code review.

## Description

#37 Added more test for syntax.py. Please review.

### CODE REVIEW

File: tests/test_syntax.py [hunk 1]
```diff
@@ -46,6 +46,42 @@ def test_python_render():
...
+        lexer_name="python",
+        line_numbers=True,
+        line_range=(2, 10),
+        theme="foo",
+        code_width=60,
+        word_wrap=True,
+        background_color="red",
+    )
+    text = syntax.highlight(CODE)
+    assert syntax.highlight(CODE) == text
```
Comment (willmcgugan) [inline tests/test_syntax.py:82]: The test is guaranteed to pass.

Comment (willmcgugan) [inline tests/test_syntax.py:82]: This test is guaranteed to pass. So not much point in having it.

File: tests/test_syntax.py [hunk 2]
```diff
@@ -46,6 +46,42 @@ def test_python_render():
...
+    )
+    from pygments.style import Style as PygmentsStyle
+
+    style = PygmentsStyle()
+    assert syntax.get_theme(style).get_background_style().bgcolor.name == "#ffffff"
+
+
+def test_pygments_syntax_theme():
+    style = PygmentsSyntaxTheme("default")
+    assert style.get_style_for_token("abc") == Style.parse("none")
```
Comment (willmcgugan) [inline tests/test_syntax.py:67]: Adds a line of coverage and the test is meaningful :-) 

There are a few more lines in `get_style_for_token` that you could check here.

File: tests/test_syntax.py [hunk 3]
```diff
@@ -46,6 +46,42 @@ def test_python_render():
...
+        line_numbers=True,
+        line_range=(2, 10),
+        theme="foo",
+        code_width=60,
+        word_wrap=True,
+    )
+    from pygments.style import Style as PygmentsStyle
+
+    style = PygmentsStyle()
+    assert syntax.get_theme(style).get_background_style().bgcolor.name == "#ffffff"
```
Comment (willmcgugan) [inline tests/test_syntax.py:62]: `get_theme` is a class method, and doesn't need an instance.

**Review by willmcgugan — CHANGES_REQUESTED** [review]
I'll need a bit more consideration re what is being tested please. The assert should be meaningful, i.e. it should fail if there is a bug.

If you could get syntax.py to 100% it would be 🥇

File: tests/test_syntax.py (hunk 1 above)
Comment (phase06) [reply tests/test_syntax.py:82]: yeah, will update the test

File: tests/test_syntax.py (hunk 2 above)
Comment (phase06) [reply tests/test_syntax.py:67]: got it

File: tests/test_syntax.py (hunk 3 above)
Comment (phase06) [reply tests/test_syntax.py:62]: will update the test

**Comment by phase06** [comment]
@willmcgugan , Got syntax.py and pretty.py to 100%. Please review 🥇

**Comment by willmcgugan** [comment]
Fantastic. Thanks!

**Pull request merged**